activesupport/lib/active_support/core_ext/module/delegation.rb

```diff
@@ -213,6 +231,9 @@ def delegate(*methods, to: nil, prefix: nil, allow_nil: nil)
 
       module_eval(method_def, file, line)
     end
+
+    method_names.each { |n| module_eval("private #{n}", file, line) } if private
```

Comment:
Should be `"private :#{n}"` or `"private #{n.inspect}"`.

Reply:
Maybe just `private(*method_names) if private`?

Reply:
@kamipo  :thinking: 

actually I've discovered that that will not work either

but this does:

Reply:
The `private` method can be passed method names.
So it just can be simplified as `send(:private, *method_names) if private`.

Reply:
I've squashed the commits to single commit, just so you know, only bit that changed is https://github.com/rails/rails/pull/31944/commits/41671930eb2b9684e5550ffc97fe9cd060afe243#diff-01f2db2585c06eca03cf014cfe51c50eR235

Reply:
@kamipo ah, ok sorry I didn't see this, I'm on it